experimental/casemap/src/casemapper.rs

```diff
@@ -142,20 +146,63 @@ impl CaseMapper {
     /// as a `LanguageIdentifier` (usually the `id` field of the `Locale`) if available, or
     /// `Default::default()` for the root locale.
     ///
-    /// See [`Self::titlecase_segment_to_string()`] for the equivalent convenience function that returns a String,
+    /// This function performs legacy head adjustment behavior when [`HeadAdjustment::Adjust`] is set. See
+    /// the docs of [`TitlecaseMapper`] for more information on what this means. There is no difference between
+    /// the behavior of this function and the equivalent ones on [`TitlecaseMapper`] when the head adjustment mode
+    /// is [`HeadAdjustment::NoAdjust`].
+    ///
+    /// See [`Self::titlecase_segment_legacy_to_string()`] for the equivalent convenience function that returns a String,
     /// as well as for an example.
-    pub fn titlecase_segment<'a>(
+    ///
+    /// [`TitlecaseMapper`]: crate::TitlecaseMapper
+    pub fn titlecase_segment_legacy<'a>(
         &'a self,
         src: &'a str,
         langid: &LanguageIdentifier,
+        options: TitlecaseOptions,
     ) -> impl Writeable + 'a {
```

Comment:
Question: Is it breaking to make this a concrete return type in the future?

Reply:
nope